full-stack-engineer/submissions/michaelyao/client/src/components/addReview/AddReview.js

```diff
@@ -0,0 +1,104 @@
+import React, { useState } from 'react';
+import {
+  Dialog,
+  Box,
+  Typography,
+  IconButton,
+  Grid,
+  Rating,
+  TextField,
+  DialogActions,
+  Button,
+  Alert,
+  Snackbar,
+} from '@mui/material';
+import CancelIcon from '@mui/icons-material/Cancel';
+import { createReview } from '../../Api';
+
+const AddReview = ({ productId, onHideAddReview }) => {
+  const [ratingValue, setRatingValue] = useState(0);
+  const [content, setContent] = useState('');
+  const [errorMessage, setErrorMessage] = useState(null);
+  const [displaySuccessAlert, setDisplaySuccessAlert] = useState(false);
+  const [reviewCreated, setReviewCreated] = useState(false);
+
+  const handleClose = () => {
+    onHideAddReview(reviewCreated);
+  };
+
+  const handleClickCreateReviewButton = async () => {
+    try {
+      await createReview(productId, ratingValue, content);
+      setDisplaySuccessAlert(true);
+      setReviewCreated(true);
+    } catch (error) {
+      setErrorMessage(error.response.data.message);
+    }
+  };
+
+  return (
+    <Dialog open onClose={handleClose} fullWidth>
+      <Box p={2}>
+        <Grid
+          container
+          direction="row"
+          justifyContent="space-between"
+          alignItems="center"
+          item
+          xs={12}
+        >
+          <Typography variant="h6" component="span">
+            New Review
+          </Typography>
+
+          <IconButton onClick={handleClose} color="error">
+            <CancelIcon />
+          </IconButton>
+        </Grid>
+        <Box mt={4}>
+          <Rating
+            name="simple-controlled"
+            value={ratingValue}
+            onChange={(event, newValue) => {
+              setRatingValue(newValue);
+            }}
+          />
+          <TextField
+            label="Content"
+            value={content}
+            onChange={(event) => setContent(event.target.value)}
+            margin="normal"
+            required
+            fullWidth
+          />
+        </Box>
+        {errorMessage && <Alert severity="error">{errorMessage}</Alert>}
```

Comment:
This is a tip for being even more pro.
Could we have a global error message IU? And call it through the context.

Reply:
um...I think it would be great for some notification system. Here the error is kind of for the page? Not sure I am correct on this or not.
<img width="1013" alt="image" src="https://user-images.githubusercontent.com/22899096/158320277-625ec24d-e112-4de0-a319-2ac22614e3d8.png">

Reply:
Yes. That would be the idea. But as this is just a Test PR and was not a requirement let's leave it like this is already great